packages/devtools_app/lib/src/charts/flutter/flame_chart.dart

```diff
@@ -286,23 +303,49 @@ abstract class FlameChartState<T extends FlameChart, V> extends State<T>
         zoomTo(math.max(FlameChart.minZoomLevel,
             zoomController.value - keyboardZoomOutUnit));
       } else if (keyLabel == 'a') {
-        scrollTo(
+        horizontallyScrollTo(
             linkedHorizontalScrollControllerGroup.offset - keyboardScrollUnit);
       } else if (keyLabel == 'd') {
-        scrollTo(
+        horizontallyScrollTo(
             linkedHorizontalScrollControllerGroup.offset + keyboardScrollUnit);
       }
     }
   }
 
   void _handlePointerSignal(PointerSignalEvent event) {
     if (event is PointerScrollEvent) {
-      if (_shiftKeyPressed) {
-        // TODO(kenz): scroll vertical list regularly / zoom. See
-        // https://github.com/flutter/devtools/issues/1600.
+      final deltaX = event.scrollDelta.dx;
+      double deltaY = event.scrollDelta.dy;
+      // TODO(kenz): shift + scroll with a mouse wheel does not register as a
+      // vertical scroll event, so we will not vertical scroll in this case (see
+      // https://github.com/flutter/flutter/issues/52767). Hopefully we can get
+      // around this if https://github.com/flutter/flutter/issues/52762 is
+      // fixed.
+      if (deltaY.abs() >= deltaX.abs()) {
+        if (_shiftKeyPressed) {
+          verticalScrollController
+              .jumpTo(verticalScrollController.offset + deltaY);
+        } else {
+          deltaY = deltaY.clamp(
+            -FlameChart.maxScrollWheelDelta,
+            FlameChart.maxScrollWheelDelta,
+          );
+          final currentZoom = zoomController.value;
+          // TODO(kenz): if https://github.com/flutter/flutter/issues/52762 is,
+          // resolved, consider adjusting the multiplier based on the scroll device
+          // kind (mouse or track pad).
+          final multiplier = FlameChart.zoomMultiplier * currentZoom;
+          final newZoomLevel = (currentZoom + deltaY * multiplier).clamp(
+            FlameChart.minZoomLevel,
+            FlameChart.maxZoomLevel,
+          );
+          zoomTo(newZoomLevel, jump: true);
```

Comment:
why is this one a jump rather than a smooth scroll?
If the frame rate allows it it would be nice to have it be smooth with a very small time interval particularly for the scroll wheel case.